controllers/openstackcluster_controller.go

```diff
@@ -682,3 +672,51 @@ func handleUpdateOSCError(openstackCluster *infrav1.OpenStackCluster, message er
 	openstackCluster.Status.FailureReason = &err
 	openstackCluster.Status.FailureMessage = pointer.String(message.Error())
 }
+
+// filterSubnets retrieves the subnets based on the Subnet filters specified on OpenstackCluster.
+func filterSubnets(networkingService *networking.Service, openStackCluster *infrav1.OpenStackCluster) ([]infrav1.Subnet, error) {
+	var subnets []infrav1.Subnet
+	openStackClusterSubnets := openStackCluster.Spec.Subnets
+	if len(openStackClusterSubnets) == 0 {
+		empty := &infrav1.SubnetFilter{}
+		listOpt := empty.ToListOpt()
+		listOpt.NetworkID = openStackCluster.Status.Network.ID
```

Comment:
`OpenStackCluster.Status.Network` is a pointer.
If `filterSubnets` is created early and the status hasn't been filled yet for some reason, this function will return the famous `panic: runtime error: invalid memory address or nil pointer dereference` error.

before the if, I'll add the following:

```golang
if openStackCluster.Status.Network == nil {
  return nil, nil
}

Reply:
the Network ID was already filled before this function was called at line 516

Reply:
Yeah you're right, because `filterSubnets` is called there as of now. I think it's fine.
If at some point we have to move the function, we'll have to be careful. With pointers I'm always adding conditions to check whether it exists, some sort of bad experiences in the past :sob: ...

Reply:
I added the protection. Thanks